crates/goose/src/agents/system.rs

```diff
@@ -15,25 +17,56 @@ pub enum SystemError {
 
 pub type SystemResult<T> = Result<T, SystemError>;
 
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Secrets {
+    /// A map of environment variables to set, e.g. API_KEY -> some_secret
```

Comment:
this right now stores the secret in plaintext. this can also be a map of `env_var_key -> key_manager_secret_name` and then we pull in the secret value and set it as env var

https://github.com/block/goose/blob/3e0ce19ddeecb6971044bf6da11a2b4b4ffa3c28/crates/goose/src/key_manager.rs#L45